### ISSUE TITLE
NewTokenClient question

### ISSUE THREAD

**Issue by lanox** [e-XpertSolutions/f5-rest-client#6]
Hi,

I have this scenario where I need to schedule the job at will run at specific time, but one of the requirements is to use token. However we use RSA as means of authentication, so the problem is if I schedule the job at 2am this will be a problem is there a way to pre-generate the token and use the token ? I hope that makes sense.

**Comment by gilliek** [comment]
Hi @lanox,

As far as I know, it is not possible to change the start time. However, it is possible to extend the lifespan of the token by modifying it with a PATCH request:

```
curl -k -vvv -u admin -X PATCH -d '{ "timeout": 12000 }' https://<BIG_URL>/mgmt/shared/authz/tokens/<TOKEN_VALUE>
```

I've made some quick tests and it seems to work, but I don't know what are the limitations. I'm not even sure if it's a bug that may be fixed in a future release, since the timeout cannot be set at the token creation.

**Comment by lanox** [comment]
Hi @gilliek 
I have hacked the package little and this is what I came up with and it works, i just dunno if you like it how I done it if so I can create  PR.

```
//NewRealTokenClient creates a new client with the given token.
func NewRealTokenClient(baseURL, token string) (*Client, error) {
	t := &http.Transport{}
	c := &Client{c: http.Client{Transport: t}, baseURL: baseURL, t: t}
	c.token = token
	c.makeAuth = authFunc(func(req *http.Request) error {
		req.Header.Add("X-F5-Auth-Token", c.token)
		return nil
	})

	return c, nil
}

// CreateToken creates a new token with the given baseURL, user, password and loginProvName.
func CreateToken(baseURL, user, password, loginProvName string) (string, error) {
	t := &http.Transport{}
	c := &Client{c: http.Client{Transport: t}, baseURL: baseURL, t: t}
	c.t.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
	// Negociate token with a pair of username/password.
	data, err := json.Marshal(map[string]string{"username": user, "password": password, "loginProviderName": loginProvName})
	if err != nil {
		return "", fmt.Errorf("failed to create token client (cannot marshal user credentials): %v", err)
	}

	tokReq, err := http.NewRequest("POST", c.makeURL("/mgmt/shared/authn/login"), bytes.NewBuffer(data))
	if err != nil {
		return "", fmt.Errorf("failed to create token client, (cannot create login request): %v", err)
	}

	tokReq.Header.Add("Content-Type", "application/json")
	tokReq.SetBasicAuth(user, password)

	resp, err := c.c.Do(tokReq)
	if err != nil {
		return "", fmt.Errorf("failed to create token client (token negociation failed): %v", err)
	}
	if resp.StatusCode >= 400 {
		return "", fmt.Errorf("failed to create token client (token negociation failed): http status %s", resp.Status)
	}
	defer resp.Body.Close()

	tok := struct {
		Token struct {
			Token string `json:"token"`
		} `json:"token"`
	}{}
	dec := json.NewDecoder(resp.Body)
	if err := dec.Decode(&tok); err != nil {
		return "", fmt.Errorf("failed to create token client (cannot decode token): %v", err)
	}
	return tok.Token.Token, nil
}


```

**Comment by lanox** [comment]
@gilliek is there any chance for patching the token ?  to be added ?

**Comment by gilliek** [comment]
Hi @lanox,

Sorry for my late reply. 

Yes, we are open to update the package to fix your problem. However, I don't understand how the code you posted solves your problem. How is `NewRealTokenClient` different from `NewTokenClient`? When you will issue your first request, the client created with NewTokenClient will negociate a token by itself.

**Comment by lanox** [comment]
well basically, I am braking the command down and returning only token, which I use as a flag.

e.g 
```script.go -token sfsfsdfsfsd89dfgdg979 -stats ```

above will use ```-token``` that will call ```NewRealTokenClien``` and ```-stats``` will get the status of pools.

the way I am doing this is little different as i need to schedule specific task to be done at specific times.  So I am making a script that you pass in the fields. I hope that make sense.

**Comment by gilliek** [comment]
Ok, I understand now and it's OK for me if you submit a PR.

Can you just change the name of the function `NewRealTokenClient` please? I'm not fond of that name, it makes the `NewTokenClient` function "not real", if you know what I mean :)

**Comment by gilliek** [comment]
Issue is solved the PR #7
